kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/PendingProduce.java

```diff
@@ -59,6 +60,7 @@ public PendingProduce(CompletableFuture<PartitionResponse> responseFuture,
             log.error("Failed to compute ByteBuf for partition '{}': {}", partitionName, e);
             return null;
         });
+        this.numberOfMessages = EntryFormatter.parseNumMessages(memoryRecords);
```

Comment:
It's duplicated because we already have a `numMessages` field.

Reply:
Ok, will remove this field and use the origin filed `numMessages `